src/transformers/configuration_utils.py

```diff
@@ -593,22 +593,21 @@ def _get_config_dict(
             local_files_only = True
 
         pretrained_model_name_or_path = str(pretrained_model_name_or_path)
-        maybe_subfolder_path = subfolder if subfolder is not None else ""
-        if os.path.isfile(os.path.join(maybe_subfolder_path, pretrained_model_name_or_path)) or is_remote_url(
+        if os.path.isfile(os.path.join(subfolder, pretrained_model_name_or_path)) or is_remote_url(
             pretrained_model_name_or_path
         ):
             config_file = pretrained_model_name_or_path
         else:
             configuration_file = kwargs.pop("_configuration_file", CONFIG_NAME)
 
-            if os.path.isdir(os.path.join(pretrained_model_name_or_path, maybe_subfolder_path)):
-                config_file = os.path.join(pretrained_model_name_or_path, maybe_subfolder_path, configuration_file)
+            if os.path.isdir(os.path.join(pretrained_model_name_or_path, subfolder)):
+                config_file = os.path.join(pretrained_model_name_or_path, subfolder, configuration_file)
             else:
                 config_file = hf_bucket_url(
                     pretrained_model_name_or_path,
                     filename=configuration_file,
                     revision=revision,
-                    subfolder=subfolder,
+                    subfolder=subfolder or None,
```

Comment:
```suggestion
                    subfolder=subfolder if len(subfolder)>0 else None,
```
I prefer explicit tests to magical `or`s :-)